### PR TITLE
[Backwards Incompatible] Use options classes for HttpHelper methods

### PR DESCRIPTION
**Pull request by tonerdo**
This PR adds option classes for the HttpHelper methods so that it can be more easily extended in a backwards compatible way. 

Addresses #902

### CODE REVIEW

File: modules/http-helper/http_helper.go [hunk 1]
```diff
@@ -16,10 +16,25 @@ import (
...
+	Headers   map[string]string
+	TlsConfig *tls.Config
+	Timeout   int
+}
+
 // HttpGet performs an HTTP GET, with an optional pointer to a custom TLS configuration, on the given URL and
 // return the HTTP status code and body. If there's any error, fail the test.
-func HttpGet(t testing.TestingT, url string, tlsConfig *tls.Config) (int, string) {
-	statusCode, body, err := HttpGetE(t, url, tlsConfig)
+func HttpGet(t testing.TestingT, options *HttpGetOptions) (int, string) {
```
Comment (brikis98) [inline modules/http-helper/http_helper.go:36]: I really like the `Options` struct approach, but these sorts of breaking changes are quite painful. Is it worth adding new methods that take in the `Options` struct and initially deprecate the old methods?

**Comment by elgatito** [comment]
@brikis98 @tonerdo I would take into consideration the idea of using `context` instead of a custom timeout and move that into a separate method, then call from old methods (with current method signatures) new methods that use contexts.

File: modules/http-helper/http_helper.go (hunk 1 above)
Comment (tonerdo) [reply modules/http-helper/http_helper.go:36]: Spending some time thinking through this

Comment (tonerdo) [reply modules/http-helper/http_helper.go:36]: @brikis98 #1078 uses your suggested approach and adds new methods that accept the options structs

**Comment by tonerdo** [comment]
Closing this in favour of #1078

**Pull request closed without merging**